### ISSUE TITLE
[BUG] Double uploads on Codecov UI

### ISSUE THREAD

**Issue by webknjaz** [codecov/feedback#613]
**Describe the bug**

Sometimes, when viewing uploads for a commit, duplicates are listed in the widget where one can inspect them.

If you look into the `Coverage reports history` widget @ https://app.codecov.io/gh/python-trio/trio/commit/4f66fabf5ca4c2dd40e4e9b76bc486a78483d159, there are 3 reports titled `macOS (pypy-3.10)`. Hovering mouse cursor on their names will reveal that one was uploaded from https://github.com/python-trio/trio/actions/runs/12392561462 (`merge_group`-triggered GHA run) and the other two were uploaded from https://github.com/python-trio/trio/actions/runs/12392781287 (a CI run triggered by a `push` event that is something that happens once merge queues succeed).

At first, I thought that maybe that second one was re-run and these could be coming from two attempts. But that's not the case — the GHA workflow run is the same.

I thought that perhaps I should download the duplicate reports and run a `diff` to see if they have the same payload. But I didn't have to go that far!
Imagine my surprise when I hovered on the `Download` links for those two duplicate reports and realized that they point to *the same URL*… Bizarre!

To summarize, it seems there's some inconsistency / DB state corruption within Codecov's backend somewhere, as it shows two entries for what clearly is the same report upload from the same CI job.

**To Reproduce**

🤷‍♂

**Expected behavior**

One upload -> one report.

**Screenshots**

https://app.codecov.io/gh/python-trio/trio/commit/4f66fabf5ca4c2dd40e4e9b76bc486a78483d159

**Additional context**

https://github.com/python-trio/trio/issues/3153#issuecomment-2551254148

**Comment by webknjaz** [comment]
@rohan-at-sentry @drazisil-codecov does the assessment seem correct?
